src/modules/flight_mode_manager/tasks/AutoFollowTarget/FlightTaskAutoFollowTarget.cpp

```diff
@@ -63,186 +67,272 @@ bool FlightTaskAutoFollowTarget::activate(const vehicle_local_position_setpoint_
 {
 	bool ret = FlightTask::activate(last_setpoint);
 
-	_position_setpoint = _position;
-
 	if (!PX4_ISFINITE(_position_setpoint(0)) || !PX4_ISFINITE(_position_setpoint(1))
 	    || !PX4_ISFINITE(_position_setpoint(2))) {
 		_position_setpoint = _position;
 	}
 
-	_target_position_filter.reset(Vector3f{NAN, NAN, NAN});
-	_offset_vector_filter.reset(Vector2f(0, 0));
-	_follow_angle_filter.reset(0.0f);
-	_velocity_ff_scale.reset(0.0f);
+	_target_pose_filter.reset(Vector3f{NAN, NAN, NAN});
+	_target_pose_filter.setParameters(TARGET_POSE_FILTER_NATURAL_FREQUENCY, TARGET_POSE_FILTER_DAMPING_RATIO);
 
-	// Initialize to something such that the drone at least points at the target, even if it's the wrong angle for the perspective.
-	// The drone will move into position as soon as the target starts moving and its heading becomes known.
-	Vector2f current_drone_heading_2d{cosf(_yaw), -sinf(_yaw)};
+	// We don't command the yawspeed, since only the yaw can be calculated off of the target
+	_yawspeed_setpoint = NAN;
```

Comment:
Current behaviour is to not set yaw at all, until drone & target's distance is bigger than minimum distance for yaw control, then sets as the raw heading from drone to target as the yaw.

Reply:
I don't believe so :thinking:. But also I thought that the sudden 360 degrees yaw spin that happens when I activate the follow me mode is from this hard setpoint, which is why I believe I want to not set it at all.

Since we don't know where the target is :thinking:

Reply:
So the FlightTask (the parent class) initializes (resets) it to NAN, so it won't be commanding at all. I guess then it makes sense to command the current yaw since commanding NAN is unpredictible? 🤔

Reply:
Yes, I would also command the current yaw :+1:

Reply:
No need to do it now, but don't forget that you're getting `last_setpoint` as a parameter, meaning that you could initialize with it (if finite) to avoid glitches (see https://github.com/PX4/PX4-Autopilot/blob/3ac8fdbe2901dcd812c8e1944f72765ddff4c022/src/modules/flight_mode_manager/tasks/Auto/FlightTaskAuto.cpp#L81)